### ISSUE TITLE
multi-node: unable to choose CNI (always applies kindnet)

### ISSUE THREAD

**Issue by sharifelgamal** [kubernetes/minikube#8222]
Right now we enforce our own CNI for multinode clusters, ignoring any input from the user. We should respect the `--enable-default-cni` parameter and not apply CNI when that's false.

**Comment by tstromberg** [comment]
The TL;DR is that CNI in minikube is very confusing.

* There is `--enable-default-cni` (default: false), which enables the CNI plug-in, but does not setup CNI. 
* There is `--network-plugin=`, required for CNI. The only option we currently support is `cni`, though there are some other more obscure alternatives in Kubernetes: https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/

Here is my rough proposal, which would also help address #1568

* Change `--enable-default-cni` (`bool`) flag to `--cni` (`string`).
  * `none`: no CNI. The current default for minikube.
  * `kindnet`: The current default for minikube if `-n` or `--driver=docker` is used.
  * `flannel`: kindnet alternative.
  * `custom`: bring your own CNI (for backwards compat, will also respect `true`).

* Add alias for `--enable-default-cni` to `--cni`.

If things look good, I also feel that `--cni=kindnet` should be the default for all clusters.
